feat(perps): integrate MetaMask Points rewards and fee discounts for Perps trading

abretonc7s opened the pull request:
## **Description**

This PR implements MetaMask Points rewards integration for Perps trading, including tier-based fee discounts and points estimation display. Users now see their estimated points earnings and receive fee discounts based on their MetaMask Points tier before executing trades.

### What is the reason for the change?
- TAT-1221: Users need tier-based MetaMask builder fee discounts (Tier 1-3: 10bps, Tier 4-5: 5bps, Tier 6-7: 3.5bps)
- TAT-1223: Users need to see estimated points earnings before trading

### What is the improvement/solution?
- Integrated RewardsController for fee discount and points estimation
- Added RewardPointsDisplay component with fox icon animation states
- Implemented fee discount visualization with fox icon in fee row
- Added comprehensive error handling with tooltips
- Supports development simulation with magic numbers (41, 42, 43)

## **Changelog**

CHANGELOG entry: Added MetaMask Points rewards integration to Perps trading with tier-based fee discounts and points estimation display

## **Related issues**

Fixes: 
- [TAT-1221](https://consensyssoftware.atlassian.net/browse/TAT-1221)
- [TAT-1223](https://consensyssoftware.atlassian.net/browse/TAT-1223)
- [TAT-1226](https://consensyssoftware.atlassian.net/browse/TAT-1426)

## **Manual testing steps**

```gherkin
Feature: MetaMask Points Rewards Integration

  Scenario: User sees fee discount for their tier
    Given user has MetaMask Points tier 4-5
    And user is on Perps order view with valid order amount
    When user enters order details
    Then user sees orange fox icon with "-X%" discount in fee row
    And total fee reflects the discount amount

  Scenario: User sees estimated points for trade
    Given user has opted into rewards program
    And user is on Perps order view
    When user enters valid order amount and leverage
    Then user sees orange fox icon with estimated points number
    And points animate on value changes (refresh state)

  Scenario: User sees error state handling
    Given rewards API is unavailable
    When user attempts to view rewards information
    Then user sees greyed out fox icon with "Couldn't load" text
    And user can tap info icon to see error details

  Scenario: Development simulation testing
    Given __DEV__ mode is enabled
    When user enters amount "41"
    Then user sees simulated 20% fee discount
    When user enters amount "42"
    Then user sees error state
    When user enters amount "43"
    Then user sees loading state
```

## **Screenshots/Recordings**

### **Before**
- No rewards integration
- Standard fees without discounts
- No points estimation

### **After**
- Fee discount displayed with fox icon and percentage
- Points estimation with animated fox icon
- Error states with tooltips
- Proper alignment and design system compliance


https://github.com/user-attachments/assets/6101166d-0565-4859-b37b-6d1569bc0afd



## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---


[TAT-1221]: https://consensyssoftware.atlassian.net/browse/TAT-1221?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

abretonc7s committed:
wip

abretonc7s committed:
feat: merge main and integrate rewards controller for perps

- Resolve merge conflicts in PerpsOrderView, perpsConfig, HyperLiquidProvider
- Migrate from direct API calls to RewardsController pattern
- Add CAIP-10 account formatting for Arbitrum perps
- Integrate fee discount display based on MetaMask Points tier
- Add points estimation for perps trading activities
- Remove direct REWARDS_API_CONFIG usage
- Keep main's cleaner logging and formatPerpsFiat utilities
- Maintain non-blocking rewards features with graceful fallbacks

abretonc7s committed:
fix: complete rewards controller migration in usePerpsOrderFees

- Remove broken REWARDS_API_CONFIG imports
- Migrate fetchFeeDiscount to use RewardsController:getPerpsDiscountForAccount
- Migrate estimatePoints to use RewardsController:estimatePoints
- Add proper CAIP-10 account formatting for Arbitrum (eip155:42161)
- Simplify EstimatePointsDto structure to match controller interface
- Maintain caching behavior and error handling
- Enable points estimation display for manual validation

This fixes the broken points estimation that was preventing users from
seeing estimated points before placing perps orders as per TASK_CALLDL_REWARDS.md

pramot999 commented on app/components/UI/Perps/hooks/usePerpsOrderFees.ts:
app/components/UI/Perps/hooks/usePerpsOrderFees.ts

abretonc7s committed:
wip

abretonc7s committed:
feat: design compatible version

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
feat: cleanup

abretonc7s committed:
resotre label guideline

abretonc7s committed:
coverage

abretonc7s committed:
feat: cleanup

abretonc7s committed:
test: enhance test coverage for perps rewards integration

- Add comprehensive edge case tests to FoxIcon.test.tsx
- Fix ESLint errors: replace 'any' types with proper typing
- Fix non-null assertion with optional chaining
- Add missing Engine mock for rewards controller
- Fix import/export style in usePerpsRewards.test.ts
- Add newline at end of test files for eslint compliance
- Complete test coverage improvements following unit testing guidelines

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
feat: cleanup

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
chore: cleanup

abretonc7s committed:
feat: refactor usePerpsOrderFees

abretonc7s committed:
cleanup

abretonc7s committed:
fix: coverage

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

nickewansmith commented:
@abretonc7s should we see the rewards in the rewards activity tab yet? After sending the order I don't see anything listed there.

abretonc7s committed:
refactor: cleanup code org

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
feat: cleanup error logs

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

VGR-GIT committed:
chore: perps discount to be based on bips

abretonc7s committed:
feat: improve codde sharing and include in closing position

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
fix: invalid fee computed on order

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
feat: coverage

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
fix: unit tests

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
Merge branch 'main' into perps/feat/rewards

abretonc7s committed:
fix: invalid test in reward controller

abretonc7s committed:
fix: test coverage

abretonc7s committed:
Merge remote-tracking branch 'origin/main' into perps/feat/rewards

abretonc7s committed:
feat: coverage
